test/e2e/create_test.go

```diff
@@ -151,6 +159,27 @@ func (tc *testContext) testBYOHConfiguration(t *testing.T) {
 	})
 }
 
+// setPowerShellDefaultShell changes the instance backed by the given Machine to have a default SSH shell of PowerShell
+func (tc *testContext) setPowerShellDefaultShell(machine *mapi.Machine) error {
+	// This needs to be retried as it will only succeed once the VM is fully initialized
```

Comment:
I don't quite understand why the retry logic is needed. I thought that since we already `waitForWindowsMachines()` to be in `Provisioned` state, the machine should already have an address by the time `setPowerShellDefaultShell` is called?